api/internal.go

```diff
@@ -130,119 +135,59 @@ func (s *PartyInfo) GetPartyInfo() {
 			continue
 		}
 
-		var encoded []byte
-		encoded, err = ioutil.ReadAll(resp.Body)
-		resp.Body.Close()
-		if err != nil {
-			log.WithField("url", rawUrl).Errorf(
-				"Unable to read partyInfo response from host, %v", err)
-			break
+		if grpc {
+			var partyInfoReq UpdatePartyInfo
```

Comment:
It would probably make sense to put these into their own functions just to shorten this method slightly

Reply:
Yes, but then there would be a lot of similarity between the two functions like mentioned before. 
We would have either a lot of code repetition if we separate them out or we like the present implementation we would have a longer code. Which would be a preferred way?